crates/shared/src/trade_finding/oneinch.rs

```diff
@@ -24,6 +27,8 @@ struct Inner {
     disabled_protocols: Vec<String>,
     protocol_cache: ProtocolCache,
     referrer_address: Option<H160>,
+    spender_cache: Arc<RwLock<Option<(H160, Instant)>>>,
```

Comment:
```suggestion
    spender_cache: Mutex<Option<(H160, Instant)>>>,
```

I think this doesn't need Arc, or is there a reason that Inner must derive Clone?

It also doesn't need RwLock because this lock won't have any contention so a plain Mutex is fine.

Reply:
> I think this doesn't need Arc, or is there a reason that Inner must derive Clone?

The `Inner` gets cloned into shared futures so it has to stay clonable.